web/src/beta/features/Dashboard/ContentsContainer/Members/index.tsx

```diff
@@ -1,66 +1,105 @@
+    currentWorkspace?.id
+  );
   const t = useT();
-  const [searchQuery, setSearchQuery] = useState<string>("");
+
+  const [searchQuery, setSearchQuery] = useState("");
   const [filteredMembers, setFilteredMembers] = useState<
     Workspace["members"] | null
   >(null);
 
-  const handleMemberSearch = () => {
-    if (!currentWorkspace?.members) {
-      return;
-    }
+  const [updatingRoleMember, setUpdatingRoleMember] = useState<TeamMember>();
+  const [addMemberModalVisible, setAddMemberModalVisible] = useState(false);
 
-    const filtered = currentWorkspace.members.filter(
-      (member) =>
-        member?.user?.name.toLowerCase().includes(searchQuery.toLowerCase()) ||
-        member?.user?.email?.toLowerCase().includes(searchQuery.toLowerCase())
-    );
-    setFilteredMembers(filtered);
-  };
+  useEffect(() => {
+    setFilteredMembers(workspace?.members || null);
+  }, [workspace]);
 
-  const handleSearchInputChange = (value: string) => {
+  const handleSearch = (value: string) => {
     setSearchQuery(value);
-    if (value === "" && currentWorkspace?.members)
-      setFilteredMembers(currentWorkspace.members);
+    if (!workspace?.members) return;
+    const v = value.toLowerCase();
+    const members = value
+      ? workspace.members.filter(({ user }) =>
+          [user?.name, user?.email].some((str) =>
+            str?.toLowerCase().includes(v)
+          )
+        )
+      : workspace.members;
+    setFilteredMembers(members);
   };
 
-  useEffect(() => {
-    if (currentWorkspace?.members) {
-      setFilteredMembers(currentWorkspace.members);
-    }
-  }, [currentWorkspace]);
+  const sortedMembers = useMemo(() => {
+    if (!filteredMembers) return [];
+    return filteredMembers.sort(
+      (a, b) =>
+        ROLE_PRIORITY[a.role] - ROLE_PRIORITY[b.role] ||
+        (a.user?.name || "").localeCompare(b.user?.name || "")
+    );
+  }, [filteredMembers]);
 
   return (
     <Wrapper>
-      <Search>
-        <TextInput
-          value={searchQuery}
-          extendWidth
-          onChange={handleSearchInputChange}
-          placeholder={t("Use Name or Email")}
-        />
-        <Button
-          icon="magnifyingGlass"
-          title={t("Find Member")}
-          size="small"
-          onClick={handleMemberSearch}
-        />
-      </Search>
+      <HeaderWrapper>
+        <Search>
+          <TextInput
+            value={searchQuery}
+            extendWidth
+            placeholder={t("Use Name or Email")}
+            onChange={handleSearch}
+          />
+          <Button
+            icon="magnifyingGlass"
+            title={t("Find Member")}
+            size="small"
+            onClick={() => handleSearch(searchQuery)}
+          />
+        </Search>
+        <div>
+          <Button
+            title={t("New member")}
+            appearance="primary"
+            icon="memberAdd"
+            onClick={() => setAddMemberModalVisible(true)}
+          />
+        </div>
```

Comment:
Only avaliable when `!config()?.disableWorkspaceManagement` and match the role rules.